marlowe/src/Language/Marlowe/Extended/V1.hs

```diff
@@ -129,9 +129,30 @@ data Contract = Close
               | When [Case] Timeout Contract
               | Let S.ValueId Value Contract
               | Assert Observation Contract
-  deriving stock (Show,Generic)
+  deriving stock (Show,Generic, Eq)
   deriving anyclass (Pretty)
 
+{-| A module is a way to package a contract with it's metadata. Eventually
+    this type could include imports and exports (reason behind the name).
+    We don't include the version number in the datatype because it is implicit
+    by the package name (`Language.Marlowe.Extended.V1`) and explicit by the
+    JSON serialization.
+-}
+data Module = Module { metadata :: MetaData
+                     -- ^ The name and information covered by this field is under discussion
+                     --   on the following thread:
+                     --   https://github.com/input-output-hk/MIPs/discussions/7
+                     --   Version 1 of Marlowe Extended shouldn't be closed until the discussion
+                     --   is resolved.
+                     , contract :: Contract
+                     -- ^ Currently we have a single entrypoint for the full contract, but we could
+                     --   extend this to be a `OMap Identifier Contract` to enable functions.
+                     --   In order to guarantee that we can convert to Marlowe Core we would need to
+                     --   validate that a function can only call functions defined above (if bottom-up)
+                     --   and that they cannot call themselves
+                     }
```

Comment:
Should we consider storing the merkleized continuations here by adding `continuations :: Map ContractHash Contract`? I realize that Extended Marlowe support merkleization, but it will be unusable for non-trivial contracts if it doesn't. Maybe now is the time to design an approach for handling merkleization of Extended Marlowe?

Reply:
I was thinking of Extended to work with the concept of functions and Core to work with the concept of merkle trees. That would make Extended more usable, right now the examples would weight the same as the unmerkelized Core.